compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/Types.java

```diff
@@ -1306,7 +1306,7 @@ public boolean isInherentlyImmutableType(BType type) {
         return false;
     }
 
-    public BType getReferredType(BType type) {
+    public static BType getReferredType(BType type) {
```

Comment:
Is there a specific reason to make this a static method in this PR?

IMO, most (if not all) methods in this class can be static methods. But given what we've been doing should we leave this as is for now?

Reply:
The reason was there can be instances where we need to extract the referred type but the type's instance cannot be retrieved as CompilerContext is not there. for example in Symbols.java. As the util method doesn't depend on CompilerContext or such; made it static. Other solution would be to duplicate the util function in such cases

Reply:
This method has been used in classes like `Symbols`, which we cannot use the instance of `Types`